bech32/src/Codec/Binary/Bech32/Internal.hs

```diff
@@ -72,40 +72,73 @@ module Codec.Binary.Bech32.Internal
 
     ) where
 
-import Prelude
```

Comment:
do you really think it is good idea?

Reply:
@paweljakubas 

The release of GHC `9.10.1` made a **_non-backwards-compatible_** change to `base`:
https://github.com/haskell/core-libraries-committee/issues/167

This leads to the following build failure, when compiled with GHC `9.10.1`:
```
[1 of 2] Compiling Codec.Binary.Bech32.Internal
src/Codec/Binary/Bech32/Internal.hs:93:1: error: [GHC-66111]
    The import of ‘Data.Foldable’ is redundant
      except perhaps to import instances from ‘Data.Foldable’
    To import instances alone, use: import Data.Foldable()
   |
93 | import Data.Foldable
   | ^^^^^^^^^^^^^^^^^^^^...
```
The error occurs because `foldl'` is now imported both by `Data.Foldable` **_and_** by `Prelude`.

But since `foldl'` is not exported by `Prelude` in versions of GHC `< 9.10`, this means that we can't
- hide `foldl'` with `import Prelude hiding (foldl')`
    (in order to hide something, it must be there in the first place)
- remove `import Data.Foldable (foldl')`
    (we need that import to get the `foldl'`)

One simple solution is to replace all implict imports with explicit imports, which should guarantee we import the same set of symbols across different GHC versions. This is the solution used by this PR.disadvantages.

Reply:
@paweljakubas Here's a similar change made in another library: https://github.com/blamario/monoid-subclasses/pull/49/commits/de0673591346a532d2a89979cc6b6d57423b8b18

Reply:
thanks for the explanation! very unfortunate change I must say ....

Reply:
> thanks for the explanation! very unfortunate change I must say ....

Yeah, I'm guessing it caused headaches for a few people!

On the bright side though, we now have `foldl'` in `Prelude`. 😄